org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/batch/FileSystem.java

```diff
@@ -167,6 +166,8 @@ public static ArrayList<Classpath> normalize(ArrayList<Classpath> classpaths) {
 	private static HashMap<File, Classpath> JRT_CLASSPATH_CACHE = null;
 	protected Map<String,Classpath> moduleLocations = new HashMap<>();
 
+	public Consumer<NameEnvironmentAnswerWithClasspathOrigin> nameEnvironmentListener = null; // listener for findType* methods
```

Comment:
We probably need a better type here and getter/setter?

Reply:
I would not make it over complicated, `Consumer<NameEnvironmentAnswer>` is OK for what is needed.
What I would add is the only public method `Consumer<NameEnvironmentAnswer> setNameEnvironmentAnswerListener(Consumer<NameEnvironmentAnswer>) and make this field private, as I don't like public fields. Return value would be original listener, if any.